### PR TITLE
Create absence request

### PR DESCRIPTION
**Pull request by tnd2908**

### CODE REVIEW

File: RoleRightApp/Controllers/RequestController.cs [hunk 1]
```diff
@@ -19,4 +20,10 @@ public async Task<List<RequestModel>> GetAllRequest()
     {
         return await _requestLogic.GetAllRequest();
     }
+
+    [HttpPost]
```
Comment (trinhhieu2908) [inline RoleRightApp/Controllers/RequestController.cs:24]: only employee can create request

**Review by hoanghieptranj5 — APPROVED** [review]

**Review by trinhhieu2908 — APPROVED** [review]

File: RoleRightApp/Logics/Abstractions/IRequestLogic.cs [hunk 2]
```diff
@@ -1,8 +1,10 @@
 ﻿using RoleRightApp.Repositories.Models;
+using RoleRightApp.RequestModels;
 
 namespace RoleRightApp.Logics.Abstractions;
 
 public interface IRequestLogic
 {
     Task<List<RequestModel>> GetAllRequest();
```
Comment (hoanghieptranj5) [inline RoleRightApp/Logics/Abstractions/IRequestLogic.cs:8]: Consider using IEnumerable<T>

File: RoleRightApp/Controllers/RequestController.cs [hunk 3]
```diff
@@ -19,4 +22,11 @@ public async Task<List<RequestModel>> GetAllRequest()
     {
         return await _requestLogic.GetAllRequest();
     }
+
+    [Authorize(Roles = Roles.Employee)]
+    [HttpPost]
+    public async Task<string> CreateRequest([FromBody] RequestRequestModel requestRequestModel) 
+    {
+        return await _requestLogic.CreateRequest(requestRequestModel);
```
Comment (hoanghieptranj5) [inline RoleRightApp/Controllers/RequestController.cs:30]: return Created(createdItemId);

File: RoleRightApp/Controllers/RequestController.cs [hunk 4]
```diff
@@ -19,4 +22,11 @@ public async Task<List<RequestModel>> GetAllRequest()
     {
         return await _requestLogic.GetAllRequest();
     }
+
+    [Authorize(Roles = Roles.Employee)]
+    [HttpPost]
+    public async Task<string> CreateRequest([FromBody] RequestRequestModel requestRequestModel)
```
Comment (hoanghieptranj5) [inline RoleRightApp/Controllers/RequestController.cs:28]: public async Task<IActionResult>

File: RoleRightApp/Repositories/Models/RequestModel.cs [hunk 5]
```diff
@@ -12,10 +12,10 @@ public class RequestModel
     public string? EmployeeId { get; set; }
 
     [DynamoDBProperty("approveBy")]
-    public string? ApproveBy { get; set; }
+    public string? ApproveBy { get; set; } = "";
```
Comment (hoanghieptranj5) [inline RoleRightApp/Repositories/Models/RequestModel.cs:15]: string.Empty

**Pull request merged**